### ISSUE TITLE
Support text 2.0

### ISSUE THREAD

**Issue by pepeiborra** [msakai/bytestring-encoding#6]
/cc @bodigrim

**Comment by Bodigrim** [comment]
```
$ cabal build --constraint 'text >= 2.0'
src/Data/ByteString/Lazy/Encoding/Internal.hs:156:30: error:
    • Couldn't match type ‘Word16’ with ‘Word8’
      Expected type: Ptr Word8
        Actual type: Ptr Word16
    • In the first argument of ‘T.fromPtr’, namely ‘workspace’
      In a stmt of a 'do' block:
        t <- T.fromPtr workspace (fromIntegral n)
      In the expression:
        do let p' :: Ptr Char
               p' = castPtr p
               ....
           n <- f (bufL buf) 0
           t <- T.fromPtr workspace (fromIntegral n)
           return ([t], buf {bufL = 0, bufR = 0})
    |
156 |               t <- T.fromPtr workspace (fromIntegral n)
    |
```

As a Hackage trustee I made revisions to prohibit unbuildable configurations:
https://hackage.haskell.org/package/bytestring-encoding-0.1.0.0/revisions/
https://hackage.haskell.org/package/bytestring-encoding-0.1.1.0/revisions/

**Comment by msakai** [comment]
Sorry to be late.
I'll take a look.

**Comment by msakai** [comment]
I released version 0.1.2.0 which supports `text` 2.0
